sys/vm/vm_page.h

```diff
@@ -875,6 +894,14 @@ vm_page_dirty(vm_page_t m)
 #endif
 }
 
+static __inline void
+vm_page_capdirty(vm_page_t m)
+{
+	if ((vm_page_astate_load(m).flags & PGA_CAPDIRTY) == 0) {
+		vm_page_aflag_set(m, PGA_CAPDIRTY);
+	}
```

Comment:
Why is this conditional? It's not for `vm_page_dirty`. But then again I've already suggested using a `vm_page_bits_t` like dirty tracking.

Reply:
It's just avoiding the RMW AMO if R indicates the bit is already set.  It's not really "conditional".